bundles/core/src/main/java/com/adobe/cq/commerce/demandware/connection/DemandwareClientImpl.java

```diff
@@ -236,9 +249,21 @@ public CloseableHttpClient getHttpClient() {
         return getHttpClientBuilder().build();
     }
 
+    @Override
+    public String getInstanceId() {
+        return instanceId;
+    }
+    
+    @Override
+    public String getAssetDownloadEndpoint(){
+        return assetDownloadEndpoint;
+    }
+
     @Activate
     protected void activate(Map<String, Object> configuration) {
         instanceEndPoint = PropertiesUtil.toString(configuration.get(INSTANCE_ENDPOINT), null);
+        assetDownloadEndpoint = PropertiesUtil.toString(configuration.get(ASSET_DOWNLOAD_ENDPOINT),"");
+        instanceId = PropertiesUtil.toString(configuration.get(INSTANCE_ID), null);
         protocolInterface = PropertiesUtil.toString(configuration.get(PROTOCOL_INTERFACE), null);
         protocolSSL = StringUtils.trimToNull(PropertiesUtil.toString(configuration.get(PROTOCOL_SSL), "TLSv1.2"));
```

Comment:
what is the default? "1.2" or "1.1" as defined in the property?

Reply:
if(instanceEndpoint == null || asset..){
	
	throw new Exception("instanceEndpoint:" + " instanceEndpoint" + ... )
}